drivers/pwm/pwm_stm32.c

```diff
@@ -678,6 +681,9 @@ static int pwm_stm32_init(const struct device *dev)
 		return r;
 	}
 
+	/* Reset timer to default state using RCC */
+	reset_line_toggle_dt(&data->reset);
```

Comment:
Wouldn't this reset _all_ PWM instances below the parent timer instance?

Reply:
Yes...
If that's an issue here, guess it also would be an issue here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/counter/counter_ll_stm32_timer.c